src/ai/backend/common/etcd.py

```diff
@@ -117,13 +121,17 @@ def _slash(v: str):
 NestedStrKeyedMapping: TypeAlias = "Mapping[str, str | NestedStrKeyedMapping]"
 NestedStrKeyedDict: TypeAlias = "dict[str, str | NestedStrKeyedDict]"
 
+T = TypeVar("T", bound=Union[EtcdClient, RaftKVSClient])
 
-class AbstractKVStore(ABC):
+
+class AbstractKVStore(ABC, Generic[T]):
     """
     Abstract interface for Key-Value Store (KVS) operations
     Defines the basic operations that a KVS should support
     """
 
+    etcd: T
```

Comment:
In my opinion, it seems better to change `etcd` to private field of EtcdClient and provide interface methods that are needed for use.

Reply:
You don't need to use `Generic` then.